File: src/main/java/com/dude/dms/ui/components/misc/ConfirmDialog.java
package com.dude.dms.ui.components.misc;

import com.vaadin.flow.component.ClickEvent;
import com.vaadin.flow.component.ComponentEventListener;
import com.vaadin.flow.component.button.Button;
import com.vaadin.flow.component.button.ButtonVariant;
import com.vaadin.flow.component.dialog.Dialog;
import com.vaadin.flow.component.html.Label;
import com.vaadin.flow.component.icon.VaadinIcon;

public class ConfirmDialog extends Dialog {

    public ConfirmDialog(String message, String confirmText, VaadinIcon icon, ComponentEventListener<ClickEvent<Button>> event, ButtonVariant theme) {
        add(new Label(message));
        Button button = new Button(confirmText, icon.create(), event);
        button.addThemeVariants(theme);
        add(button);
    }
}

File: src/main/java/com/dude/dms/ui/components/misc/ConfirmDialog.kt
package com.dude.dms.ui.components.misc

import com.dude.dms.ui.components.dialogs.DmsDialog
import com.vaadin.flow.component.ClickEvent
import com.vaadin.flow.component.ComponentEventListener
import com.vaadin.flow.component.button.Button
import com.vaadin.flow.component.button.ButtonVariant
import com.vaadin.flow.component.icon.VaadinIcon

class ConfirmDialog(
        message: String,
        confirmText: String,
        icon: VaadinIcon,
        theme: ButtonVariant,
        event: ComponentEventListener<ClickEvent<Button>>
) : DmsDialog(message) {

    init {
        add(Button(confirmText, icon.create(), event).apply {
            addThemeVariants(theme)
            addClickListener { close() }
        })
    }
}